src/V3Width.cpp

```diff
@@ -1377,6 +1377,38 @@ class WidthVisitor final : public VNVisitor {
         }
     }
 
+    void visit(AstSetuphold* nodep) override {
+        FileLine* const flp = nodep->fileline();
+
+        AstAssignW* newp = nullptr;
+
+        if (nodep->delrefp() != nullptr) {
+            AstNodeVarRef* lhsp = VN_AS(nodep->delrefp()->cloneTreePure(false), VarRef);
+            lhsp->access(VAccess::WRITE);
+            AstNodeVarRef* rhsp = VN_AS(nodep->refevp()->cloneTreePure(false), VarRef);
```

Comment:
Need error if the user didn't put a varref as one of the arguments, and need a test for that error.  Similar below (and nullptr/const cleanup too).

```suggestion
        if (nodep->delrefp()) {
            AstNodeVarRef* const lhsp = VN_AS(nodep->delrefp()->cloneTreePure(false), VarRef);
            lhsp->access(VAccess::WRITE);
            AstNodeVarRef* const rhsp = VN_AS(nodep->refevp()->cloneTreePure(false), VarRef);
```

Reply:
Added an internal error in case the type is incorrect. I couldn't think of a test for that though. The parser seems to handle most references that are incorrect, like function identifiers, numbers etc. Also this transformation happens very early, and it seems there are no Ref transformations that could affect this.